include/xrpl/protocol/TxFlags.h

```diff
@@ -121,13 +121,7 @@ constexpr std::uint32_t tfClearDeepFreeze                  = 0x00800000;
 constexpr std::uint32_t tfTrustSetMask =
     ~(tfUniversal | tfSetfAuth | tfSetNoRipple | tfClearNoRipple | tfSetFreeze |
       tfClearFreeze | tfSetDeepFreeze | tfClearDeepFreeze);
-
-// valid flags for granular permission
-constexpr std::uint32_t tfTrustSetGranularMask = tfSetfAuth | tfSetFreeze | tfClearFreeze;
-
-// bits representing supportedGranularMask are set to 0 and the bits
-// representing other flags are set to 1 in tfPermissionMask.
-constexpr std::uint32_t tfTrustSetPermissionMask = (~tfTrustSetMask) & (~tfTrustSetGranularMask);
+constexpr std::uint32_t tfTrustSetPermissionMask = ~(tfUniversal | tfSetfAuth | tfSetFreeze | tfClearFreeze);
```

Comment:
I see removing `tfTrustSetMask` from `tfTrustSetPermissionMask`, meaning only the three listed here operations can be explicitly delegated by `TrustSet` and the previous mask was in error. Do I read this right ?

Reply:
With the previous flags, the unit-tests to test the granular permission for invalid flags should have succeeded but they failed. It's worth investigating why they failed.

Reply:
The check in the implementation is `if (txFlags & tfMPTokenIssuanceSetPermissionMask)`  and then in the tests `tfFullyCanonicalSig` was added to the flags in the transaction. So because `tfUniversal` didn't exist, it failed. So thats why the flags needed to be update. I think its the same for TrustSet. I was also curious.